account_banking_pain_base/models/account_payment_order.py

```diff
@@ -115,30 +121,41 @@ def _compute_sepa(self):
         eur = self.env.ref("base.EUR")
         sepa_list = self._sepa_iban_prefix_list()
         for order in self:
-            sepa = True
-            if order.company_partner_bank_id.acc_type != "iban":
-                sepa = False
-            if (
-                order.company_partner_bank_id
-                and order.company_partner_bank_id.sanitized_acc_number[:2]
-                not in sepa_list
-            ):
-                sepa = False
-            for pline in order.payment_line_ids:
-                if pline.currency_id != eur:
-                    sepa = False
-                    break
-                if pline.partner_bank_id.acc_type != "iban":
+            sepa_payment_method = False
```

Comment:
For making the code more readable, since v13 you can assign directly to the field the variable without performance penalty, and also assign for the whole recordset.